### PR TITLE
Update the mempool app to v2.5.0

### PR DESCRIPTION
**Pull request by knorrium**
This update brings the mempool app to parity with the production version. If LND is installed, the Lightning Explorer will be enabled.

The new app has been tested on an Ubuntu VM and Raspberry Pi running Umbrel 0.5.3.

Note: Additional testing by the Umbrel team is advised due to the community app store changes. This update has been tested using a modified Umbrel install and a temporary community app store repository, as discussed by another user in #358. See also https://github.com/getumbrel/umbrel/issues/1564#issuecomment-1373150661

### CODE REVIEW

**Comment by wiz** [comment]
https://github.com/mempool/mempool/releases/tag/v2.5.0

**Comment by nmfretz** [comment]
Thanks @knorrium and @wiz, I've been looking forward to this. I'll review and test on amd64 and arm64 soon.

>Note: Additional testing by the Umbrel team is advised due to the community app store changes.

Noted. Thanks for this.

**Comment by wiz** [comment]
@lukechilds can you guys please test and merge this PR ?

**Review by nmfretz — COMMENTED** [review]
Sorry for the delay on this @knorrium and @wiz. I got hit with a cold late last week.

Really great work on this. I love the new Lightning explorer... an absolute ton of work went into it.

I have tested on arm64 and amd64 and it works well. I also checked that the following install sequence works: 
`user has Bitcoin Node and Electrs installed --> installs Mempool --> installs Lightning Node`

^Under this sequence Lightning Node related env vars are set to blank strings when mempool is first installed, meaning that if the user later installs Lightning Node, they will need to restart their Umbrel (or Mempool app directly) in order to set the env vars correctly. Only then will Lightning explorer functionality come online. See review below for suggestions on noting this in the app store description.

Pinging @lukechilds for another set of eyes here because we may want to add logic to make sure that the `/lightning` app-data directory is created as user 1000:1000 for the case when Lightning Node isn't installed. Here is what currently happens with the `Bitcoin Node and Electrs installed --> install Mempool --> install Lightning Node` flow:

Docker creates `${UMBREL_ROOT}/app-data/lightning/data/lnd` with root ownership:

```
$ ls -l
total 16
drwxr-xr-x 5 admin admin 4096 Mar 30 18:01 bitcoin
drwxr-xr-x 4 admin admin 4096 Apr  3 05:59 electrs
drwxr-xr-x 3 root  root  4096 Apr  3 18:41 lightning
drwxr-xr-x 4 admin admin 4096 Apr  3 05:58 mempool
```

Later, when the Lightning Node app is installed, `${UMBREL_ROOT}/app-data/lightning/` ownership is changed to 1000:1000.

```
a$ ls -l
total 16
drwxr-xr-x 5 admin admin 4096 Mar 30 18:01 bitcoin
drwxr-xr-x 4 admin admin 4096 Apr  3 05:59 electrs
drwxr-xr-x 4 admin admin 4096 Apr  3 18:46 lightning
drwxr-xr-x 4 admin admin 4096 Apr  3 05:58 mempool

$ echo "UID: $(id -u), GID: $(id -g)"
UID: 1000, GID: 1000
```

So on the surface there appears to be no issues with having the empty lightning data dir owned by root when no Lightning Node app is installed. What do you think @lukechilds? Is there a security implication I'm missing or a case where this could cause issues when installing the Lightning Node app for some users?

File: mempool/umbrel-app.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ description: >-
 
   - JSON APIs
```
[the remark this answers is not in the document]
Comment (lukechilds) [reply mempool/umbrel-app.yml:33]: It looks like TV view is still available at `Graphs > TV View`, it's just no longer in the main menu.

File: mempool/umbrel-app.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ manifestVersion: 1
 id: mempool
 category: Explorers
 name: mempool
-version: "2.4.0-build-2"
+version: "2.5.0"
 tagline: A self-hosted explorer for the Bitcoin community
 description: >-
```
[the remark this answers is not in the document]
Comment (lukechilds) [reply mempool/umbrel-app.yml:7]: Yeah agree we should add something like this, I like this working. @wiz @knorrium what do you think?

**Comment by lukechilds** [comment]
>What do you think @lukechilds? Is there a security implication I'm missing, or a case where this could cause issues when installing the Lightning Node app for some users? Binding ${UMBREL_ROOT}/app-data/ would solve the ownership issue but reduce security.

Nope, this is all looking great! Just tested locally. The permissions get fixed by `rsync` at install time of the Lightning Node app because the `data/lnd/` path exists in the app repo template and `rsync` will ensure the source/destination permissions match even if the destination path already exists.

One thing to watch out for would be if we ever remove this path from the app template in the future, it would then be left behind as root in the install Bitcoin > Mempool > Lightning flow. But that's probably not gonna happen and the alternative of giving Mempool access to all apps is less secure as you mentioned above so I think the approach we have here makes the most sense.

File: mempool/docker-compose.yml [hunk 3]
```diff
@@ -16,18 +16,22 @@ services:
...
-    image: mempool/backend:v2.4.0@sha256:6036cc6866b782efadc765e8189958940202b3fa1770eb35c35210c86efd2f47
+    image: mempool/backend:v2.5.0@sha256:c58996a3bfd382cb782da5d93e4b8f65f9c02c7a6bb10df4dda0bdf52ba8ed88
     user: "1000:1000"
     init: true
     restart: on-failure
     stop_grace_period: 1m
     command: "./wait-for-it.sh mariadb:3306 --timeout=720 --strict -- ./start.sh"
     volumes:
       - ${APP_DATA_DIR}/data:/backend/cache
+      - ${UMBREL_ROOT}/app-data/lightning/data/lnd:/lnd:ro
```
[the remark this answers is not in the document]
Comment (lukechilds) [reply mempool/docker-compose.yml:34]: This will be correctly handled by `rsync` here: https://github.com/getumbrel/umbrel/blob/f60bebecae30de37a41633514e339be7ea1914c0/scripts/app#L409 assuming [`data/lnd`](https://github.com/getumbrel/umbrel-apps/tree/112169469178b9e7a106f66c7d32826a07a4bcd6/lightning/data/lnd) exists in the app repo template of the version of Lightning Node app that's being installed.

File: mempool/umbrel-app.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmfretz) [reply mempool/umbrel-app.yml:33]: whoops, good catch. Look at me trying to remove mempool features...

**Comment by nmfretz** [comment]
Perfect. Thanks, @lukechilds.

Okay, @knorrium this can go live as soon as these two small comments are addressed. 
- https://github.com/getumbrel/umbrel-apps/pull/474#discussion_r1156310255
- https://github.com/getumbrel/umbrel-apps/pull/474#discussion_r1156303057

File: mempool/umbrel-app.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (knorrium) [reply mempool/umbrel-app.yml:7]: Just updated the description with a more concise wording.

File: mempool/umbrel-app.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (knorrium) [reply mempool/umbrel-app.yml:37]: Added the Lightning explorer to to the list of features.

File: mempool/docker-compose.yml [hunk 4]
```diff
@@ -46,6 +50,14 @@ services:
       MEMPOOL_CLEAR_PROTECTION_MINUTES: "20"
       MEMPOOL_INDEXING_BLOCKS_AMOUNT: "52560"
       MEMPOOL_STDOUT_LOG_MIN_PRIORITY: "info"
+      LIGHTNING_ENABLED: "true"
+      LIGHTNING_BACKEND: "lnd"
+      LIGHTNING_STATS_REFRESH_INTERVAL: 3600
+      LIGHTNING_GRAPH_REFRESH_INTERVAL: 3600
+      LND_TLS_CERT_PATH: "/lnd/tls.cert"
+      LND_MACAROON_PATH: "/lnd/data/chain/bitcoin/$APP_BITCOIN_NETWORK/readonly.macaroon"
+      LND_REST_API_URL: "https://$APP_LIGHTNING_NODE_IP:$APP_LIGHTNING_NODE_REST_PORT"
```
[the remark this answers is not in the document]
Comment (knorrium) [reply mempool/docker-compose.yml:59]: @lukechilds @nmfretz have you folks considered a way to restart the apps from the web UI? Perhaps the `Manage apps` state could include restarting the app as an option in addition to removing them. The app store entry could be an option too but that might not be as straightforward to users as the main home screen.

**Comment by wiz** [comment]
ship it

**Comment by nmfretz** [comment]
Shipped. Ping me if you would like to update any of the gallery images to include the lightning explorer.

File: mempool/umbrel-app.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nmfretz) [reply mempool/umbrel-app.yml:7]: Thanks

File: mempool/docker-compose.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (nmfretz) [reply mempool/docker-compose.yml:59]: Yep, this will be a feature in the future for sure 🤫
